interfaces/builtin/greengrass_support.go

```diff
@@ -126,31 +241,58 @@ mount options=(rw, bind) /proc/irq/ -> /proc/irq/,
 mount options=(rw, bind) /proc/sys/ -> /proc/sys/,
 mount options=(rw, bind) /proc/sysrq-trigger -> /proc/sysrq-trigger,
 
-# remap a few /proc accesses to /dev/null
+# mount some devices using /dev/null
 mount options=(rw, bind) /dev/null -> /proc/kcore,
 mount options=(rw, bind) /dev/null -> /proc/sched_debug,
 mount options=(rw, bind) /dev/null -> /proc/timer_stats,
 
-# perform the pivot_root into the overlay
-pivot_root oldroot=/var/snap/greengrass/@{SNAP_REVISION}/rootfs/.pivot_root*/ /var/snap/greengrass/*/rootfs/,
-mount options=(rw, rprivate) -> /.pivot_root*/,
-umount /.pivot_root*/,
-owner /.pivot_root*/ w,
-mount options=(rw, rprivate) -> /,
-mount options=(ro, remount, rbind) -> /,
+# umounts for tearing down containers
+umount /var/snap/{@{SNAP_NAME},@{SNAP_INSTANCE_NAME}}/*/**,
+
+# this is for container device creation
+# also need mknod and mknodat in seccomp
+capability mknod,
 
-# allow tearing down the overlay
-umount /var/snap/@{SNAP_NAME}/**,
-/run/mount/utab rw,
-/bin/umount ixr,
+# for the greengrassd pid file
+# note we can't use layouts for this because /var/run is a symlink to /run
+# and /run is explictly disallowed for use by layouts
+# also note that technically this access is post-pivot_root, but during the setup
+# for the mount ns that the snap performs (not snapd), /var/run is bind mounted
+# from outside the pivot_root to inside the pivot_root, so this will always 
+# access the same files inside or outside the pivot_root
+owner /{var/,}run/greengrassd.pid rw,
 
-# For lambda functions, post pivot_root lambda execution accesses
+# all of the rest of the accesses are made by child containers and as such are 
+# "post-pivot_root", meaning that they aren't accessing these files on the 
+# host root filesystem, but rather somewhere inside $SNAP_DATA/rootfs/
+# Note: eventually greengrass will gain the ability to specify child profiles
+# for it's containers and include these rules in that profile so they won't
+# be here, but that work isn't done yet
+# Additionally see LP bug #1791711 for apparmor resolving file accesses after
+# a pivot_root
+
+# for child container lambda certificates
 /certs/ r,
 /certs/** r,
 /group/ r,
 /group/** r,
 /state/ r,
-/state/sqlite* rwk,
+/state/{,**} r,
+# the child containers need to use a file lock here
+owner /state/secretsmanager/secrets.db krw,
+owner /state/secretsmanager/secrets.db-journal rw,
+owner /state/shadow/ rw,
+owner /state/shadow/{,**} krw,
+# more specific accesses for writing
+owner /state/server/ rw,
+owner /state/server/{,**} rw,
+
+# for executing python, nodejs, java, and C (executable) lambda functions
+# currently the runtimes are "python2.7", "nodejs6.10", "java8" and "executable",
+# but those version numbers could change so we add a "*" on the end of the folders to be safe for
+# future potential upgrades
+/runtime/{python*,executable*,nodejs*,java*}/ r,
```

Comment:
I would rather leave it as is, simply because this is technically a leak if the host filesystem for some odd reason has things inside `/runtime` I think it's better to only support specific runtimes.